Sources/Basics/Archiver+Tar.swift

```diff
@@ -0,0 +1,147 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import class Dispatch.DispatchQueue
+import struct Dispatch.DispatchTime
+import struct TSCBasic.AbsolutePath
+import protocol TSCBasic.FileSystem
+import struct TSCBasic.FileSystemError
+import class TSCBasic.Process
+
+/// An `Archiver` that handles Tar archives using the command-line `tar` tool.
+public final class TarArchiver: Archiver {
```

Comment:
I dont think this should be a reference type. It holds no state

Reply:
It holds references to `Cancellator` and `FileSystem`, both of which are reference types. `Cancellator` in particular holds mutable state with a cancellation handlers registry.

Reply:
sure, but that does not mean it also needs to be a reference type